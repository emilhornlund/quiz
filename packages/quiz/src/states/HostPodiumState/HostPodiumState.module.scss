@import '../../styles/helpers';

.main {
  display: flex;
  flex-direction: column;
  height: 100%;
  flex: 1;

  @include mobile {
    row-gap: $layout-gutter;
    width: 100%;
  }

  @include tablet {
    row-gap: $layout-gutter * 1.5;
    width: $mobile-width;
  }

  @include desktop {
    row-gap: $layout-gutter * 2;
    width: $tablet-width;
  }

  .title {
    display: flex;
    flex-direction: row;
    justify-content: center;
    font-weight: bold;
    text-align: center;
    color: white;
    width: 100%;

    @include mobile {
      @include fontSize(18px);
    }

    @include tablet {
      @include fontSize(28px);
    }

    @include desktop {
      @include fontSize(38px);
    }
  }
}
