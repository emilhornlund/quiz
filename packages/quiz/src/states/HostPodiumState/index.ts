export type { HostPodiumStateProps } from './HostPodiumState'
export { default } from './HostPodiumState'
