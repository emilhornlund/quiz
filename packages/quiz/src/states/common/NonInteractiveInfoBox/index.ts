export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export { default } from './NonInteractiveInfoBox'
