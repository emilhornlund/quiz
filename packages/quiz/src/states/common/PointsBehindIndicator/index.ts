export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export { default } from './PointsBehindIndicator'
