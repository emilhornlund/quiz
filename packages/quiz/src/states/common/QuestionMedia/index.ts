export type { QuestionMediaProps } from './QuestionMedia'
export { default } from './QuestionMedia'
