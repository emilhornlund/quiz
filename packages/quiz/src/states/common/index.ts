export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export type { QuestionAnswerPickerProps } from './QuestionAnswerPicker'
export type { QuestionMediaProps } from './QuestionMedia'
export type { QuestionTextPreviewProps } from './QuestionTextPreview'
export type { QuestionTypePointsBarProps } from './QuestionTypePointsBar'

export { default as GamePage } from './GamePage'
export { default as NonInteractiveInfoBox } from './NonInteractiveInfoBox'
export { default as PointsBehindIndicator } from './PointsBehindIndicator'
export { default as QuestionAnswerPicker } from './QuestionAnswerPicker'
export { default as QuestionMedia } from './QuestionMedia'
export { default as QuestionTextPreview } from './QuestionTextPreview'
export { default as QuestionTypePointsBar } from './QuestionTypePointsBar'
