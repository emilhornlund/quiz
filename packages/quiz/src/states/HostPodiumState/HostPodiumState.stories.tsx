import { GameEventType } from '@quiz/common'
import type { Meta, StoryObj } from '@storybook/react'

import HostPodiumState from './HostPodiumState'

const meta = {
  component: HostPodiumState,
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof HostPodiumState>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    event: {
      type: GameEventType.PodiumHost,
      leaderboard: [
        { position: 1, nickname: 'ShadowCyborg', score: 18456 },
        { position: 2, nickname: 'Radar', score: 18398 },
        { position: 3, nickname: 'ShadowWhirlwind', score: 15492 },
        { position: 4, nickname: 'WhiskerFox', score: 14118 },
        { position: 5, nickname: 'JollyNimbus', score: 13463 },
        { position: 6, nickname: 'PuddingPop', score: 12459 },
        { position: 7, nickname: 'MysticPine', score: 11086 },
        { position: 8, nickname: 'FrostyBear', score: 10361 },
        { position: 9, nickname: 'Willo', score: 9360 },
        { position: 10, nickname: 'ScarletFlame', score: 6723 },
      ],
    },
  },
} satisfies Story

export const ThreePlayers = {
  args: {
    event: {
      type: GameEventType.PodiumHost,
      leaderboard: [
        { position: 1, nickname: 'ShadowCyborg', score: 18456 },
        { position: 2, nickname: 'Radar', score: 18398 },
        { position: 3, nickname: 'ShadowWhirlwind', score: 15492 },
      ],
    },
  },
} satisfies Story
