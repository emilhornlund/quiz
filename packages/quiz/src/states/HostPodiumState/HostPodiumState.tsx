import { GameEventPodiumHost } from '@quiz/common'
import React, { FC } from 'react'

import { Leaderboard, Page, Podium } from '../../components'

import styles from './HostPodiumState.module.scss'

export interface HostPodiumStateProps {
  event: GameEventPodiumHost
}

const HostPodiumState: FC<HostPodiumStateProps> = ({
  event: { leaderboard },
}) => (
  <Page>
    <div className={styles.main}>
      <div className={styles.title}>Podium</div>
      <Podium values={leaderboard} />
      <Leaderboard values={leaderboard} includePodium={false} />
    </div>
  </Page>
)

export default HostPodiumState
