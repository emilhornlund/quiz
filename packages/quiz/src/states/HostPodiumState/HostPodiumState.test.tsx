import { GameEventType } from '@quiz/common'
import { render } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { describe, expect, it } from 'vitest'

import HostPodiumState from './HostPodiumState'

describe('HostPodiumState', () => {
  it('should render HostPodiumState', async () => {
    const { container } = render(
      <MemoryRouter>
        <HostPodiumState
          event={{
            type: GameEventType.GamePodiumHost,
            leaderboard: [
              { position: 1, nickname: 'ShadowCyborg', score: 18456 },
              { position: 2, nickname: 'Radar', score: 18398 },
              { position: 3, nickname: 'ShadowWhirlwind', score: 15492 },
              { position: 4, nickname: 'WhiskerFox', score: 14118 },
              { position: 5, nickname: 'JollyNimbus', score: 13463 },
              { position: 6, nickname: 'PuddingPop', score: 12459 },
              { position: 7, nickname: 'MysticPine', score: 11086 },
              { position: 8, nickname: 'FrostyBear', score: 10361 },
              { position: 9, nickname: 'Willo', score: 9360 },
              { position: 10, nickname: 'ScarletFlame', score: 6723 },
            ],
          }}
          onComplete={() => undefined}
        />
      </MemoryRouter>,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render HostPodiumState without a full leaderboard', async () => {
    const { container } = render(
      <MemoryRouter>
        <HostPodiumState
          event={{
            type: GameEventType.GamePodiumHost,
            leaderboard: [
              { position: 1, nickname: 'ShadowCyborg', score: 18456 },
              { position: 2, nickname: 'Radar', score: 18398 },
              { position: 3, nickname: 'ShadowWhirlwind', score: 15492 },
            ],
          }}
          onComplete={() => undefined}
        />
      </MemoryRouter>,
    )

    expect(container).toMatchSnapshot()
  })
})
