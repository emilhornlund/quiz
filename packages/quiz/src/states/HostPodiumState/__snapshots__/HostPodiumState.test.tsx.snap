// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HostPodiumState > clicks Quit and calls completeTask 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <div
          class="buttonInputContainer buttonInputKindPrimary buttonInputSizeSmall"
        >
          <button
            data-testid="test-fullscreen-button-button"
            id="fullscreen-button"
            name="fullscreen-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-maximize"
              data-icon="maximize"
              data-prefix="fas"
              role="img"
              viewBox="0 0 448 512"
            >
              <path
                d="M168 32L24 32C10.7 32 0 42.7 0 56L0 200c0 9.7 5.8 18.5 14.8 22.2S34.1 223.8 41 217l40-40 79 79-79 79-40-40c-6.9-6.9-17.2-8.9-26.2-5.2S0 302.3 0 312L0 456c0 13.3 10.7 24 24 24l144 0c9.7 0 18.5-5.8 22.2-14.8s1.7-19.3-5.2-26.2l-40-40 79-79 79 79-40 40c-6.9 6.9-8.9 17.2-5.2 26.2S270.3 480 280 480l144 0c13.3 0 24-10.7 24-24l0-144c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2l-40 40-79-79 79-79 40 40c6.9 6.9 17.2 8.9 26.2 5.2S448 209.7 448 200l0-144c0-13.3-10.7-24-24-24L280 32c-9.7 0-18.5 5.8-22.2 14.8S256.2 66.1 263 73l40 40-79 79-79-79 40-40c6.9-6.9 8.9-17.2 5.2-26.2S177.7 32 168 32z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="buttonInputContainer buttonInputKindCallToAction buttonInputSizeSmall"
        >
          <button
            data-testid="test-skip-button-button"
            disabled=""
            id="skip-button"
            name="skip-button"
            type="button"
          >
            <div
              class="loadingSpinner"
            >
              <div />
              <div />
              <div />
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="content centerAlign"
    >
      <div
        class="contentWrapper mediumWidth"
      >
        <div
          class="contentInner"
        >
          <div
            class="typography subtitle full"
          >
            Podium
          </div>
          <div
            class="main"
          >
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Beta
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  2
                </div>
                <div
                  class="score"
                >
                  90
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Alpha
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  1
                </div>
                <div
                  class="score"
                >
                  100
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Gamma
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  3
                </div>
                <div
                  class="score"
                >
                  80
                </div>
              </div>
            </div>
          </div>
          <div
            class="main"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HostPodiumState > clicks Quit when completeTask is undefined 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <div
          class="buttonInputContainer buttonInputKindPrimary buttonInputSizeSmall"
        >
          <button
            data-testid="test-fullscreen-button-button"
            id="fullscreen-button"
            name="fullscreen-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-maximize"
              data-icon="maximize"
              data-prefix="fas"
              role="img"
              viewBox="0 0 448 512"
            >
              <path
                d="M168 32L24 32C10.7 32 0 42.7 0 56L0 200c0 9.7 5.8 18.5 14.8 22.2S34.1 223.8 41 217l40-40 79 79-79 79-40-40c-6.9-6.9-17.2-8.9-26.2-5.2S0 302.3 0 312L0 456c0 13.3 10.7 24 24 24l144 0c9.7 0 18.5-5.8 22.2-14.8s1.7-19.3-5.2-26.2l-40-40 79-79 79 79-40 40c-6.9 6.9-8.9 17.2-5.2 26.2S270.3 480 280 480l144 0c13.3 0 24-10.7 24-24l0-144c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2l-40 40-79-79 79-79 40 40c6.9 6.9 17.2 8.9 26.2 5.2S448 209.7 448 200l0-144c0-13.3-10.7-24-24-24L280 32c-9.7 0-18.5 5.8-22.2 14.8S256.2 66.1 263 73l40 40-79 79-79-79 40-40c6.9-6.9 8.9-17.2 5.2-26.2S177.7 32 168 32z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="buttonInputContainer buttonInputKindCallToAction buttonInputSizeSmall"
        >
          <button
            data-testid="test-skip-button-button"
            disabled=""
            id="skip-button"
            name="skip-button"
            type="button"
          >
            <div
              class="loadingSpinner"
            >
              <div />
              <div />
              <div />
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="content centerAlign"
    >
      <div
        class="contentWrapper mediumWidth"
      >
        <div
          class="contentInner"
        >
          <div
            class="typography subtitle full"
          >
            Podium
          </div>
          <div
            class="main"
          >
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Beta
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  2
                </div>
                <div
                  class="score"
                >
                  90
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Alpha
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  1
                </div>
                <div
                  class="score"
                >
                  100
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Gamma
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  3
                </div>
                <div
                  class="score"
                >
                  80
                </div>
              </div>
            </div>
          </div>
          <div
            class="main"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HostPodiumState > renders subtitle and leaderboard entries 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <div
          class="buttonInputContainer buttonInputKindPrimary buttonInputSizeSmall"
        >
          <button
            data-testid="test-fullscreen-button-button"
            id="fullscreen-button"
            name="fullscreen-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-maximize"
              data-icon="maximize"
              data-prefix="fas"
              role="img"
              viewBox="0 0 448 512"
            >
              <path
                d="M168 32L24 32C10.7 32 0 42.7 0 56L0 200c0 9.7 5.8 18.5 14.8 22.2S34.1 223.8 41 217l40-40 79 79-79 79-40-40c-6.9-6.9-17.2-8.9-26.2-5.2S0 302.3 0 312L0 456c0 13.3 10.7 24 24 24l144 0c9.7 0 18.5-5.8 22.2-14.8s1.7-19.3-5.2-26.2l-40-40 79-79 79 79-40 40c-6.9 6.9-8.9 17.2-5.2 26.2S270.3 480 280 480l144 0c13.3 0 24-10.7 24-24l0-144c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2l-40 40-79-79 79-79 40 40c6.9 6.9 17.2 8.9 26.2 5.2S448 209.7 448 200l0-144c0-13.3-10.7-24-24-24L280 32c-9.7 0-18.5 5.8-22.2 14.8S256.2 66.1 263 73l40 40-79 79-79-79 40-40c6.9-6.9 8.9-17.2 5.2-26.2S177.7 32 168 32z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="buttonInputContainer buttonInputKindCallToAction buttonInputSizeSmall"
        >
          <button
            data-testid="test-skip-button-button"
            id="skip-button"
            name="skip-button"
            type="button"
          >
            <span>
              Quit
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-right"
              data-icon="arrow-right"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
            >
              <path
                d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-105.4 105.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="content centerAlign"
    >
      <div
        class="contentWrapper mediumWidth"
      >
        <div
          class="contentInner"
        >
          <div
            class="typography subtitle full"
          >
            Podium
          </div>
          <div
            class="main"
          >
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Beta
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  2
                </div>
                <div
                  class="score"
                >
                  90
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Alpha
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  1
                </div>
                <div
                  class="score"
                >
                  100
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Gamma
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  3
                </div>
                <div
                  class="score"
                >
                  80
                </div>
              </div>
            </div>
          </div>
          <div
            class="main"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HostPodiumState > should render HostPodiumState 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <div
          class="buttonInputContainer buttonInputKindPrimary buttonInputSizeSmall"
        >
          <button
            data-testid="test-fullscreen-button-button"
            id="fullscreen-button"
            name="fullscreen-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-maximize"
              data-icon="maximize"
              data-prefix="fas"
              role="img"
              viewBox="0 0 448 512"
            >
              <path
                d="M168 32L24 32C10.7 32 0 42.7 0 56L0 200c0 9.7 5.8 18.5 14.8 22.2S34.1 223.8 41 217l40-40 79 79-79 79-40-40c-6.9-6.9-17.2-8.9-26.2-5.2S0 302.3 0 312L0 456c0 13.3 10.7 24 24 24l144 0c9.7 0 18.5-5.8 22.2-14.8s1.7-19.3-5.2-26.2l-40-40 79-79 79 79-40 40c-6.9 6.9-8.9 17.2-5.2 26.2S270.3 480 280 480l144 0c13.3 0 24-10.7 24-24l0-144c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2l-40 40-79-79 79-79 40 40c6.9 6.9 17.2 8.9 26.2 5.2S448 209.7 448 200l0-144c0-13.3-10.7-24-24-24L280 32c-9.7 0-18.5 5.8-22.2 14.8S256.2 66.1 263 73l40 40-79 79-79-79 40-40c6.9-6.9 8.9-17.2 5.2-26.2S177.7 32 168 32z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="buttonInputContainer buttonInputKindCallToAction buttonInputSizeSmall"
        >
          <button
            data-testid="test-skip-button-button"
            id="skip-button"
            name="skip-button"
            type="button"
          >
            <span>
              Quit
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-right"
              data-icon="arrow-right"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
            >
              <path
                d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-105.4 105.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="content centerAlign"
    >
      <div
        class="contentWrapper mediumWidth"
      >
        <div
          class="contentInner"
        >
          <div
            class="typography subtitle full"
          >
            Podium
          </div>
          <div
            class="main"
          >
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Radar
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  2
                </div>
                <div
                  class="score"
                >
                  18398
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  ShadowCyborg
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  1
                </div>
                <div
                  class="score"
                >
                  18456
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  ShadowWhirlwind
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  3
                </div>
                <div
                  class="score"
                >
                  15492
                </div>
              </div>
            </div>
          </div>
          <div
            class="main"
          >
            <div
              class="column"
            >
              <div
                class="position"
              >
                <span>
                  4
                </span>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="row"
              >
                <span>
                  WhiskerFox
                </span>
              </div>
            </div>
            <div
              class="column"
            >
              <span
                class="score"
              >
                14118
              </span>
            </div>
            <div
              class="column"
            >
              <div
                class="position"
              >
                <span>
                  5
                </span>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="row"
              >
                <span>
                  JollyNimbus
                </span>
              </div>
            </div>
            <div
              class="column"
            >
              <span
                class="score"
              >
                13463
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HostPodiumState > should render HostPodiumState without a full leaderboard 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <div
          class="buttonInputContainer buttonInputKindPrimary buttonInputSizeSmall"
        >
          <button
            data-testid="test-fullscreen-button-button"
            id="fullscreen-button"
            name="fullscreen-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-maximize"
              data-icon="maximize"
              data-prefix="fas"
              role="img"
              viewBox="0 0 448 512"
            >
              <path
                d="M168 32L24 32C10.7 32 0 42.7 0 56L0 200c0 9.7 5.8 18.5 14.8 22.2S34.1 223.8 41 217l40-40 79 79-79 79-40-40c-6.9-6.9-17.2-8.9-26.2-5.2S0 302.3 0 312L0 456c0 13.3 10.7 24 24 24l144 0c9.7 0 18.5-5.8 22.2-14.8s1.7-19.3-5.2-26.2l-40-40 79-79 79 79-40 40c-6.9 6.9-8.9 17.2-5.2 26.2S270.3 480 280 480l144 0c13.3 0 24-10.7 24-24l0-144c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2l-40 40-79-79 79-79 40 40c6.9 6.9 17.2 8.9 26.2 5.2S448 209.7 448 200l0-144c0-13.3-10.7-24-24-24L280 32c-9.7 0-18.5 5.8-22.2 14.8S256.2 66.1 263 73l40 40-79 79-79-79 40-40c6.9-6.9 8.9-17.2 5.2-26.2S177.7 32 168 32z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
        <div
          class="buttonInputContainer buttonInputKindCallToAction buttonInputSizeSmall"
        >
          <button
            data-testid="test-skip-button-button"
            id="skip-button"
            name="skip-button"
            type="button"
          >
            <span>
              Quit
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-right"
              data-icon="arrow-right"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
            >
              <path
                d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-105.4 105.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="content centerAlign"
    >
      <div
        class="contentWrapper mediumWidth"
      >
        <div
          class="contentInner"
        >
          <div
            class="typography subtitle full"
          >
            Podium
          </div>
          <div
            class="main"
          >
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  Radar
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  2
                </div>
                <div
                  class="score"
                >
                  18398
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  ShadowCyborg
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  1
                </div>
                <div
                  class="score"
                >
                  18456
                </div>
              </div>
            </div>
            <div
              class="column"
            >
              <div
                class="spacer"
              />
              <div
                class="spacer"
              />
              <div
                class="nickname"
              >
                <div
                  class="main"
                >
                  ShadowWhirlwind
                </div>
              </div>
              <div
                class="stack"
              >
                <div
                  class="position"
                >
                  3
                </div>
                <div
                  class="score"
                >
                  15492
                </div>
              </div>
            </div>
          </div>
          <div
            class="main"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
